java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java

```diff
@@ -529,7 +529,7 @@ private void determineEncoding() {
       // fallback to DIRECT encoding.
       // The decision to use patched base was based on zigzag values, but the
       // actual patching is done on base reduced literals.
-      if ((brBits100p - brBits95p) != 0) {
+      if ((brBits100p - brBits95p) != 0 && Math.abs(min) < 1l << 56) {
```

Comment:
can you make `(1L << 56)` as static final constant?

Reply:
Can you also include unit test case that was mentioned in the description of the ticket?